### ISSUE TITLE
ScalafmtModule hardcodes scala version

### ISSUE THREAD

**Issue by nh13** [lihaoyi/mill#442]
https://github.com/lihaoyi/mill/blob/146d58bc5d98da1d142758f974d4d5de6f5948bf/scalalib/src/mill/scalalib/scalafmt/ScalafmtModule.scala#L27

I suppose we could have `ScalafmtModule` extend `ScalaModule` and use `scalaVersion()`?  

My use case is trying to use `def scalafmtVersion = "1.6.0-RC4"`, which requires scala `2.12.6`, as well as overriding `scalafmtDeps` as follows to include the reflect library:

```
  def scalafmtDeps: T[Agg[PathRef]] = T {
    Lib.resolveDependencies(
      zincWorker.repositories,
      Lib.depToDependency(_, scalaVersion()),
      Seq(ivy"com.geirsson::scalafmt-cli:${scalafmtVersion()}", ivy"org.scala-lang:scala-reflect:${scalaVersion()}")
    )
  }
```

I'd be happy to submit a PR if folks agree on the fix.

**Comment by nh13** [comment]
It would also be nice to have scalafmt run on compile, like in sbt.

**Comment by guilgaly** [comment]
In addition to the Scala version thing, the group ID has also changed for Scalafmt 2.x, from `com.geirsson` to `org.scalameta`. If nobody's already working on it, I'll try to write a quick PR that makes it easier to upgrade to the latest RC version of Scalafmt.

Another interesting idea would be to support `scalafmt-dynamic`, which allows to simply specify the Scalafmt version in the Scalafmt configuration file; but that would probably take more work.

**Comment by guilgaly** [comment]
BTW, in the mean time, a work-around allowing to use the latest Scalafmt 2.0 RC is to use something like this in a build:
```scala
trait MyScalafmtModule extends ScalafmtModule {
  // We need to override this (not just 'scalafmtVersion') because the group ID has changed
  override def scalafmtDeps: T[Agg[PathRef]] = T {
    Lib.resolveDependencies(
      zincWorker.repositories,
      Lib.depToDependency(_, "2.12.8"),
      Seq(ivy"org.scalameta::scalafmt-cli:2.0.0-RC5"),
    )
  }
}
```

**Comment by lefou** [comment]
Fixed by #600
